src/type/CardState.type.ts

```diff
@@ -0,0 +1,9 @@
+export type CardState = {
+  cardNumber: string[];
+  expirationDate: string[];
+  ownerName: string;
+  securityCode: string;
+  label: string;
+  color: string;
+  description: string;
+};
```

Comment:
```suggestion
type CardNumber = `${number}${number}${number}${number}`;
type ExpirationMonth = `${0}${number}` | `1${0 | 1 | 2}`
type ExpirationYear = `${number}${number}`

export type CardState = {
  cardNumber: [CardNumber, CardNumber, CardNumber, CardNumber];
  expirationDate: [ExpirationMonth, ExpirationYear];
  ownerName: string;
  securityCode: `${numbe}${number}${number}`;
  label: string;
  color: string;
  description: string;
};
```

이런식으로 타입을 더 정교하게 작성할 수 있답니다!

Reply:
> 이런식으로 타입을 더 정교하게 작성할 수 있답니다!

😲 Array tuple 타입을 알고 있었는데, 저렇게 string type도 명시적으로 쓸 수 있는지 몰랐어요! 👍👍👍